CONTRIBUTING.md

```diff
@@ -0,0 +1,128 @@
+
+```
+git remote add upstream https://github.com/Real-Dev-Squad/website-goals
+```
+
+4. **Getting the latest code from the develop branch** (Can be skipped if you've cloned the repo just now)
+
+If it's been quite a while after you have cloned the repo/made the last pull request, it's recommended to take a pull from the develop branch. Reason being, there may be some changes which could have merged after you had cloned the repo/made the last pull request.
+
+To do so, make sure you're in the develop branch by checking out to the **develop** branch:
+
+```
+git checkout develop
+```
+
+Once you're in the **develop** branch, it's time to take a pull:
+
+```
+git pull upstream develop
+```
+
+Now that you've made sure that you've got latest changes, we can proceed to creating our branch
+
+5. **Creating a branch**
+
+Create a new branch to work on. We require a different branch so that we always have a stable, working version in the default (develop) branch. We're not supposed to touch the **main** branch as it is the one getting deployed on production.
+
+```
+git checkout -b <branch-name>
+```
+
+We will try to name the branch according to the task we are going to perform in it. If it is going to be a `feature`, the branch name should begin with `feat` or `feature`. If it is going to be a `fix`, the branch name should begin with `fix` or `bugfix`. The branch name should be self-explanatory.
+For example, if I want to work on a `feature` called `login-form`, the branch name will be **feature/login-form**. If it is going to be a `fix` in `form-data`, the branch name will be `fix/form-data`.
+Command example:
+
+```
+git checkout -b feature/login-form
+```
+
+6. **Just do it!**
+
+Perform the tasks you wanted to, can be anything, ranging from fixing simple typo to re-designing the whole page!
+
+7. **Committing your changes**
+
+Now you have made the changes, though they are saved in your system, Git doesn't know what changes you've done. So you have to **commit** your changes. First step is to add the files which you want to add to the staging area, the dot after **add** in the first command tells Git to check for changes in all the files. The second step is about committing your changes. The message part is short description of your commit, what changes that particular commit is doing. For example, if while creating a login form, the commit message can be something like `creating login form`.
+
+```
+git add .
+git commit -m "Write message about your commit"
+```
+
+8. **Making sure you have the latest changes from the develop branch**
+
+It may so happen that since the last time you cloned the repo/took a pull from develop, some changes may be merged in the develop branch. So to be on the safer side, we should have those changes as well.
+
+In order to do that, we first checkout to **develop** branch by:
+
+```
+git checkout develop
+```
+
+Once we're in develop, it's time to take a pull:
+
+```
+git pull upstream develop
+```
+
```

Comment:
Don't you think this section is getting repeated in pt. 4. Can we shorten it and make it more comprehensive in one point itself. What you think? or Is it really required.

Reply:
I just added it because, it was the samein every other repos as well. I think for a beginner not knowing this might create an issue, so let it be, this is what I think.

Reply:
Yeah i have noticed in other repos too . but won't it be better if you just provide a reference to point 4 instead of repeating the same. what you say?
for eg. Like repeat step 4